app/controllers/service_controller.rb

```diff
@@ -256,6 +256,9 @@ def get_node_info(treenodeid, _show_list = true)
     @nodetype, id = parse_nodetype_and_id(valid_active_node(treenodeid))
     # resetting action that was stored during edit to determine what is being edited
     @sb[:action] = nil
+
+    @nodetype, id = parse_nodetype_and_id(params[:id]) if params[:id]
```

Comment:
@AparnaKarve : I understand you need this behavior added here, but it creates another hidden data path in the code.

What I mean: now the input to `get_node_info` can be passed as an argument `treenodeid` or (newly) through `params[:id]`.

That is a bad design and it's going to be hard to fix in the future.

Can you move this logic to `show`? That is where you get with the click, right?

Reply:
> Can you move this logic to show? That is where you get with the click, right?

@martinpovolny I understand, that is the most logical thing to do -- only if it worked.

If you check the Service explorer code, you will notice that `show` redirects to `explorer`
and then you get this -
```
F, [2017-11-03T07:54:49.876401 #79870] FATAL -- : Error caught: [NoMethodError] undefined method `name' for nil:NilClass
~/manageiq/plugins/manageiq-ui-classic/app/controllers/service_controller.rb:292:in `get_node_info'
~/manageiq/plugins/manageiq-ui-classic/app/controllers/application_controller.rb:2365:in `set_active_elements'
~/manageiq/plugins/manageiq-ui-classic/app/controllers/application_controller.rb:2383:in `build_accordions_and_trees'
~/manageiq/plugins/manageiq-ui-classic/app/controllers/service_controller.rb:70:in `explorer'
```

The whole `show` method looks really questionable to me.

The `nodetype` evaluation happens in `get_node_info` - where the above check that I have added really matters - since `show` is going to eventually end up there via `explorer`

Unfortunately, most explorers are just poor quality code, full of bad surprises - precisely the reason why I wanted to stay away from Explorer in Generic Objects.

Without agitating the Service Explorer code too much (as in , a refactor), I think what I have here is our best bet.

Reply:
> Unfortunately, most explorers are just poor quality code, full of bad surprises - precisely the reason why I wanted to stay away from Explorer in Generic Objects.

Well unless people actually start fixing it, it will shay poor quality code.

Adding functionality is an opportunity to cleanup stuff and remove exceptions. Because to add functionality you read the code and see what is bad. Well unless you use try-test-fail approach that actually leads to even worse code ;-)

Reply:
> Adding functionality is an opportunity to cleanup stuff and remove exceptions. 

Sure, but not at this given time.
It's an upstream effort. 
A refactor going in for the Release, is just plain risky and not a good idea.